Source/Android/src/org/dolphinemu/dolphinemu/AssetCopyService.java

```diff
@@ -36,7 +36,11 @@ public AssetCopyService()
 	@Override
 	protected void onHandleIntent(Intent intent)
 	{
-		String BaseDir = Environment.getExternalStorageDirectory() + File.separator + "dolphin-emu";
+		String BaseDir;
+		final String DefaultDir = Environment.getExternalStorageDirectory() + File.separator + "dolphin-emu";
+		if ((BaseDir = NativeLibrary.GetUserDirectory()) == "")
```

Comment:
Couldn't you just do the assignment to the variable directy and then do `if (BaseDir.isEmpty())` as the conditional?